File: src/2287.rearrange-characters-to-make-target-string.2372/solution.ts
function rearrangeCharacters (s: string, target: string): number {
    const countS = count(s);
    const countT = count(target);
    let result = s.length;
    for (let i = 0; i < 26; i++) {
        if (countT[i] === 0) {
            continue;
        }
        result = Math.min(result, (countS[i] / countT[i]) | 0);
    }
    return result;
}

function count (s:string) {
    const result = new Array<number>(26).fill(0);
    for (const c of s) {
        result[c.charCodeAt(0) - 97]++;
    }
    return result;
}

File: src/2287.rearrange-characters-to-make-target-string.2372/solution.java
class Solution {
    public int rearrangeCharacters(String s, String target) {
        int[] count1 = toCount(s);
        int[] count2 = toCount(target);
        int result = Integer.MAX_VALUE;
        for(int i=0;i<26;i++){
            if(count2[i] == 0){
                continue;
            }
            result = Math.min(result, count1[i]/count2[i] );

        }
        return result;
    }

    private int[] toCount(String s){
        int[] result = new int[26];
        for(char c:s.toCharArray()){
            result[c-'a']++;
        }
        return result;
    }
}